ProcessesApi/V1/Gateways/SoleToJointGateway.cs

```diff
@@ -0,0 +1,139 @@
+
+            var result = await _dynamoDbContext.LoadAsync<PersonDbEntity>(id).ConfigureAwait(false);
+            return result?.ToDomain();
+        }
+
+        [LogCall]
+        private async Task<PaymentAgreement> GetPaymentAgreementByTenureId(Guid tenureId, Guid correlationId)
+        {
+            _logger.LogDebug($"Calling Income API for payment agreeement with Tenure ID: {tenureId}");
+            var route = $"{_apiGateway.ApiRoute}/agreements/{tenureId}";
+            return await _apiGateway.GetByIdAsync<PaymentAgreement>(route, tenureId, correlationId);
+        }
+
+        [LogCall]
+        private async Task<Tenancy> GetTenancyById(Guid id, Guid correlationId)
+        {
+            _logger.LogDebug($"Calling Income API with tenancy ID: {id}");
+            var route = $"{_apiGateway.ApiRoute}/tenancies/{id}";
+            return await _apiGateway.GetByIdAsync<Tenancy>(route, id, correlationId);
+        }
+
+        public async Task<bool> CheckTenureFinanceRecords(Guid tenureId)
+        {
+            var paymentAgreement = await GetPaymentAgreementByTenureId(tenureId, Guid.NewGuid()).ConfigureAwait(false); // TODO: Confirm what correlation ID to use
+            if (paymentAgreement != null && paymentAgreement.Amount > 0)
+            {
+                return false;
+            }
+            else
+            {
+                var tenancy = await GetTenancyById(tenureId, Guid.NewGuid()).ConfigureAwait(false); // TODO: Confirm what correlation ID to use
+                if (tenancy is null)
+                    return true; // throw error?
+
+                if (tenancy.nosp.active)
+                    return false;
+                return true;
+            }
+
+        }
+
+        public async Task<bool> CheckPersonTenureRecord(Guid tenureId, Guid proposedTenantId)
+        {
+            var tenure = await GetTenureById(tenureId).ConfigureAwait(false);
+            if (tenure is null)
+                return true;// throw error?
+
+            var personHouseholdMemberRecord = tenure.HouseholdMembers.ToListOrEmpty().Find(x => x.Id == proposedTenantId);
+            if (personHouseholdMemberRecord is null)
+                return true;// throw error?
+
+            if (tenure.TenureType.Code != TenureTypes.Secure.Code ||
+                (tenure.HouseholdMembers.Count(x => x.IsResponsible) > 1 && personHouseholdMemberRecord.IsResponsible))
+            {
+                return false;
+            }
+            else
+            {
+                return await CheckTenureFinanceRecords(tenureId).ConfigureAwait(false);
+            }
+        }
+
+        public async Task<bool> CheckEligibility(Guid tenureId, Guid proposedTenantId)
+        {
+            var currentTenure = await GetTenureById(tenureId).ConfigureAwait(false);
+            if (currentTenure is null)
+                return false; // TODO: Confirm whether should raise error 
```

Comment:
We could get it to throw an error with a message of  something like "No tenure with this Id"